components/public-api/gitpod/v1/prebuild.proto

```diff
@@ -0,0 +1,107 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+import "gitpod/v1/pagination.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service PrebuildService {
+    rpc StartPrebuild(StartPrebuildRequest) returns (StartPrebuildResponse) {}
+    rpc StopPrebuild(StopPrebuildRequest) returns (StopPrebuildResponse) {}
+
+    rpc GetPrebuild(GetPrebuildRequest) returns (GetPrebuildResponse) {}
+    rpc ListPrebuilds(ListPrebuildsRequest) returns (ListPrebuildsResponse) {}
+    rpc WatchPrebuild(WatchPrebuildRequest) returns (stream WatchPrebuildResponse) {}
+}
+
+message GetPrebuildRequest {
+    string prebuild_id = 1;
+    string workspace_id = 2;
```

Comment:
This is a bit ambiguous: Does it mean: get the prebuild that a) "this workspace is based on" or b) "that was produced by this workspace"?
I _feel_ b) should be less common with the new API (maybe we can remove it entirely? Did not check if we still use it, though). But at least a comment would be helpful here. If we still need to support both cases, we could also think about two different fields. But I hope we can avoid that.

Reply:
Yes, let's treat prebuilds as a resource on its own and shield the fact that we internally reuse workspaces.

Reply:
But I'm unser somewhat 🤔 we [can discuss in sync/ or on slack](https://gitpod.slack.com/archives/C0463A2KQUU/p1699888057629959).